registration-processor/workflow-engine/registration-processor-workflow-manager-service/src/test/java/io/mosip/registration/processor/workflow/manager/stage/WorkflowInternalActionVerticleTest.java

```diff
@@ -0,0 +1,250 @@
+
+	@Mock
+	WebSubUtil webSubUtil;
+
+
+	@InjectMocks
+	private WorkflowInternalActionVerticle workflowInternalActionVerticle = new WorkflowInternalActionVerticle() {
+		@Override
+		public MosipEventBus getEventBus(Object verticleName, String url, int instanceNumber) {
+			vertx = Vertx.vertx();
+
+			return new MosipEventBus() {
+
+				@Override
+				public Vertx getEventbus() {
+					return vertx;
+				}
+
+				@Override
+				public void consume(MessageBusAddress fromAddress,
+						EventHandler<EventDTO, Handler<AsyncResult<MessageDTO>>> eventHandler) {
+
+				}
+
+				@Override
+				public void consumeAndSend(MessageBusAddress fromAddress, MessageBusAddress toAddress,
+						EventHandler<EventDTO, Handler<AsyncResult<MessageDTO>>> eventHandler) {
+
+				}
+
+				@Override
+				public void send(MessageBusAddress toAddress, MessageDTO message) {
+
+				}
+			};
+		}
+
+		@Override
+		public void consumeAndSend(MosipEventBus eventbus, MessageBusAddress addressbus1,
+				MessageBusAddress addressbus2, long messageExpiryTimeLimit) {
+		}
+
+		@Override
+		public Router postUrl(Vertx vertx, MessageBusAddress consumeAddress, MessageBusAddress sendAddress) {
+			return null;
+
+		}
+
+		@Override
+		public void createServer(Router router, int port) {
+
+		}
+	};
+
+	@Test
+	public void testDeployVerticle() {
+
+		ReflectionTestUtils.setField(workflowInternalActionVerticle, "workerPoolSize", 10);
+		ReflectionTestUtils.setField(workflowInternalActionVerticle, "clusterManagerUrl", "/dummyPath");
+		workflowInternalActionVerticle.deployVerticle();
+	}
+
+	@Test
+	public void testStart() {
+		ReflectionTestUtils.setField(workflowInternalActionVerticle, "port", "2333");
+		Mockito.doNothing().when(router).setRoute(any());
+		workflowInternalActionVerticle.start();
+	}
+
+	@Test
+	public void testProcessSuccessForMarkAsPaused() {
+		WorkflowInternalActionDTO workflowInternalActionDTO = new WorkflowInternalActionDTO();
+		workflowInternalActionDTO.setRid("10006100390000920200603070407");
+		workflowInternalActionDTO.setActionCode(WorkflowInternalActionCode.MARK_AS_PAUSED.toString());
+		workflowInternalActionDTO.setActionMessage("packet is paused");
+		workflowInternalActionDTO.setResumeTimestamp("2021-03-02T08:24:29.526Z");
+		workflowInternalActionDTO.setEventTimestamp("2021-03-02T08:24:29.526Z");
+		workflowInternalActionDTO.setDefaultResumeAction("ResumeProcessing");
+		Mockito.doNothing().when(registrationStatusService).updateRegistrationStatus(any(), any(), any());
+		registrationStatusDto = new InternalRegistrationStatusDto();
+		registrationStatusDto.setRegistrationId("10006100390000920200603070407");
+		Mockito.when(auditLogRequestBuilder.createAuditRequestBuilder(any(), any(), any(), any(), any(), any(), any()))
+				.thenReturn(null);
+		Mockito.when(registrationStatusService.getRegistrationStatus(anyString())).thenReturn(registrationStatusDto);
+		workflowInternalActionVerticle.process(workflowInternalActionDTO);

```

Comment:
We should use argument capture for websubutil and registrationStatusService.updateRegistrationStatus and validate to ensure the dto is modified and called back.

This applies for other 4 test cases also for each actions

Ref: QualityClassifierStageTest.java for Argument capture

Reply:
fixed